analysis/vertfn/analyzer.go

```diff
@@ -66,45 +87,147 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	printer = &SortedPrinter{Pass: pass, Printer: printer}
 	defer printer.Flush()
 
-	inspect.Preorder([]ast.Node{&ast.FuncDecl{}, &ast.CallExpr{}}, func(n ast.Node) {
-		if fn, ok := n.(*ast.FuncDecl); ok && fn != nil {
-			fnDecl[fnameFromFuncDecl(fn)] = fn
+	check := func(ref *ast.Ident, def token.Pos, kind RefKind) {
+		if !def.IsValid() {
+			// So far only seen on calls to Error method of error interface
+			printer.Info(ref.Pos(), "got invalid definition position")
+			return
+		}
+		// Ignore references to definitions that aren't at file scope, e.g. to local variables
+		defFileScope := pass.Pkg.Scope().Innermost(pass.Fset.File(def).LineStart(1))
+		defScope := pass.Pkg.Scope().Innermost(def)
+		if defScope != defFileScope {
+			printer.Info(ref.Pos(), fmt.Sprintf("skpping definition with inner scope from %s", pass.Fset.Position(defScope.Pos())))
+			return
 		}
 
-		if call, ok := n.(*ast.CallExpr); ok && call != nil {
-			for _, fname := range fnameFromCallExpr(call) {
-				fnCall[fname] = append(fnCall[fname], call)
-			}
+		if RefOrder[kind] == Ignore {
+			printer.Info(ref.Pos(), fmt.Sprintf("%s reference %s ignored by options", kind, ref.Name))
 		}
-	})
 
-	for fn, def := range fnDecl {
-		fnDeclFile := pass.Fset.File(def.Pos()).Name()
-		fnDeclLineNum := pass.Fset.Position(def.Pos()).Line
+		if pass.Fset.File(ref.Pos()).Name() != pass.Fset.File(def).Name() {
+			printer.Ok(ref.Pos(), fmt.Sprintf(`%s reference %s is to definition in separate file (%s)`, kind, ref.Name, pass.Fset.Position(def)))
```

Comment:
for separate file I am using now info. let's keep it as info to not stand out of green vs red.
```suggestion
			printer.Info(ref.Pos(), fmt.Sprintf(`%s reference %s is to definition in separate file (%s)`, kind, ref.Name, pass.Fset.Position(def)))
```